inline UI helper text and tracking (source mode)

amyleew opened the pull request:
Adding inline helper text to Studio UI in source mode - specifically for minzoom, maxzoom and buffer definition and application.

![image](https://cloud.githubusercontent.com/assets/4587826/6149974/7d956600-b1da-11e4-9a0c-b4fd28854b82.png)

![image](https://cloud.githubusercontent.com/assets/4587826/6149995/95adb9a4-b1da-11e4-9b18-e21fcd013ed2.png)

Closes #1027
Closes #1026

amyleew committed:
first pass

amyleew commented:
Other options for more copy and scalability.

### Click reveal

![expand_click](https://cloud.githubusercontent.com/assets/4587826/6150233/2df08f42-b1dc-11e4-865b-ec031c9ab997.gif)

### Open docs tab

![image](https://cloud.githubusercontent.com/assets/4587826/6150127/7af0c3e4-b1db-11e4-8f14-23fc7acc3120.png)

![04_docs-tab](https://cloud.githubusercontent.com/assets/4587826/6150244/43bcbbd4-b1dc-11e4-8d84-fc2ff2b65c6b.png)

samanpwbb commented:
I definitely agree that inline text is the approach we should take at least to start. Looks great to me. Make sure you update all the source templates!

amyleew commented:
Updated copy - take a look before I roll out to other formats @ajashton @samanpwbb @yhahn 

![image](https://cloud.githubusercontent.com/assets/4587826/6153616/d82d59ec-b1f3-11e4-9e6f-e33f763a72ea.png)

![image](https://cloud.githubusercontent.com/assets/4587826/6153622/e94d5de4-b1f3-11e4-8cfa-dc77cbf29c1c.png)

amyleew committed:
copy edits

amyleew committed:
more changes added tracking

amyleew committed:
adding tracking

amyleew committed:
fixed link

amyleew committed:
change maxzoom in ui to 16 not 22

amyleew committed:
add all tracking to source save

amyleew commented:
Added [on source SAVE tracking](https://github.com/mapbox/mapbox-studio/blob/1df75ff6ac49699440afefd9f7f8ddf16ce2e33c/app/source.js#L539-L551) to:
- layer count
- buffer size
- max and minzoom settings
- file size

Changed both min and maxzoom to max out at 16 vs. 22:

![image](https://cloud.githubusercontent.com/assets/4587826/6157097/e4ecdfa4-b20c-11e4-84b8-302f3c0ed35c.png)

/cc @yhahn @ajashton @samanpwbb

yhahn commented on app/source.js:
```diff
@@ -534,6 +535,21 @@ window.Source = function(templates, cwd, tm, source, revlayers, examples, isMapb
+
+		// Track file size on save
+		analytics.track('file size', { filesize: filesize });
+		
```
want to consistify the indenting here? Studio is using 4 spaces in many places for tabs.

yhahn commented on app/source.js:
```diff
@@ -574,13 +590,13 @@ window.Source = function(templates, cwd, tm, source, revlayers, examples, isMapb
             tiles: ['/source/{z}/{x}/{y}.png?id=' + this.model.id + '&' + mtime],
             minzoom: this.model.get('minzoom'),
-            maxzoom: 22
+            maxzoom: 16
```
These maxzoom values controls the max zoom that the source editor allows you to preview your source at. I think it probably makes sense to leave this at z22 (ie. so you can see what your source data looks like overzoomed).

To limit the maxzoom that the sliders in the settings pane allow I think you'll want to adjust the max properties here:

https://github.com/mapbox/mapbox-studio/blob/mb-pages/templates/source.html#L91-L104

amyleew committed:
tab fix, remove preview z16 constraints

amyleew committed:
fix spacing

amyleew committed:
fix spacing

yhahn commented on app/source.js:
```diff
@@ -534,6 +535,21 @@ window.Source = function(templates, cwd, tm, source, revlayers, examples, isMapb
+
+        // Track file size on save
+        analytics.track('file size', { filesize: filesize });
+		
```
Per chat, 
- These will need to be above the `return` on line 537 to get called.
- `attr.layer.buffer-size` won't work in js -- you'll want to access it with the brackets + quotes (http://stackoverflow.com/a/7122629)

amyleew commented:
@yhahn Finally got it tracking minzoom, maxzoom and buffer sizes per layers on zoom. thanks @samanpwbb's tutelage on console.log and objects 

![image](https://cloud.githubusercontent.com/assets/4587826/6160059/d30be8a6-b22b-11e4-8b1a-31d640445bb9.png)

![image](https://cloud.githubusercontent.com/assets/4587826/6160063/d768d1b6-b22b-11e4-8906-f5ab98ebcbf9.png)

amyleew committed:
proper coding for tracking buffers and min/maxzooms

amyleew committed:
Merge branch 'mb-pages' of github.com:mapbox/mapbox-studio into ui-updates

amyleew committed:
spaces and console notes removed

amyleew committed:
replace lines

amyleew committed:
add changelog info

yhahn commented:
:ship:

yazjisuhail commented:
@amyleew why did you guys lock the max zoom to 16 in the latest version?

yazjisuhail commented:
@amyleew modifying the max zoom in the source project's yml won't load the data beyond 16. does it need any kind of refresh/restart?

amyleew commented:
@yazjisuhail Close Studio and use a text editor (like Sublime) to modify the `maxzoom` in the `data.yml` in your project's .tm2source folder. Restart Studio and your file's `maxzoom` will be changed.
